### DISCUSSION TITLE
TAC or NPV as part of the objective function

### DISCUSSION THREAD

**Discussion by ayp234** [FZJ-IEK3-VSA/FINE#56] [Q&A] [answered]
Hello,

I noticed that you mention multiple times that your objective function is about minimizing the TAC of the whole system.However, in the declareObjective function of the energySystemModel class, you do not mention TAC, but NPV. This method uses the getEconomicsDesign method of the component class (via the getObjectiveFunctionContribution method of the component). The default setting of getEconomicsDesign is  getOptValueCostType="TAC". So, it is about minimizing the TAC after all? Is it possible to switch to NPV?

**Comment by JohannesBehrens** [comment, marked as answer]
The objective function of the current fine version is about the minimization of the total system NPV. This was introduced together with the implementation of multi-period optimization (version >= 2.3.0). Previously, the objective function was constructed from the TAC. The default for getEconomicsDesign was set to TAC to ensure that the function could be used if a user used the function directly in his code to calculate the TAC of a component within an already optimized system and does not pass the newly introduced argument. However, in this case (single year optimization), the TAC and NPV of the calculation are the same, since only one investment can be made (at the beginning of the period) which does not need to be discounted afterwards. Whenever the function is used to calculate the contribution to the objective function, the NPV is used.
